src/check/check_types/unify.zig

```diff
@@ -1,153 +1,3590 @@
+    /// **Case 2: `a` Extends `b`**
+    ///
+    /// a = { x, y, z }ext_a
+    /// b = { x, y }ext_b
+    ///
+    /// - `a` has additional fields not in `b`
+    /// - We generate a new var `only_in_a_var = { z }ext_a`
+    /// - Unify `only_in_a_var ~ ext_b`
+    /// - Then unify shared fields
+    ///
+    /// ---
+    ///
+    /// **Case 3: `b` Extends `a`**
+    ///
+    /// a = { x, y }ext_a
+    /// b = { x, y, z }ext_b
+    ///
+    /// - Same as Case 2, but reversed
+    /// - `b` has additional fields not in `a`
+    /// - We generate a new var `only_in_b_var = { z }ext_b`
+    /// - Unify `ext_a ~ only_in_b_var`
+    /// - Then unify shared fields
+    ///
+    /// ---
+    ///
+    /// **Case 4: Both Extend Each Other**
+    ///
+    /// a = { x, y, z }ext_a
+    /// b = { x, y, w }ext_b
+    ///
+    /// - Each has unique fields the other lacks
+    /// - Generate:
+    ///     - shared_ext = fresh flex_var
+    ///     - only_in_a_var = { z }shared_ext
+    ///     - only_in_b_var = { w }shared_ext
+    /// - Unify:
+    ///     - `ext_a ~ only_in_b_var`
+    ///     - `only_in_a_var ~ ext_b`
+    /// - Then unify shared fields into `{ x, y }shared_ext`
+    ///
+    /// ---
+    ///
+    /// All field unification is done using `unifySharedFields`, and new variables are created using `fresh`.
+    ///
+    /// This function does not attempt to deduplicate fields or reorder them — callers are responsible
+    /// for providing consistent field names.
+    fn unifyTwoRecords(
+        self: *Self,
+        vars: *const ResolvedVarDescs,
+        a_record: Record,
+        b_record: Record,
+    ) error{TypeMismatch}!void {
+
+        // First, unwrap all fields for record a, panicaing with various non-recoverable error
+        // These pancis will likely be changed/removed in the future
+        const a_gathered_fields = try self.gatherRecordFields(a_record);
+        const b_gathered_fields = try self.gatherRecordFields(b_record);
+
+        // Then partition the fields
+        const partitioned = partitionFields(
+            &self.types_store.env.idents,
+            self.scratch,
+            a_gathered_fields.range,
+            b_gathered_fields.range,
+        );
+
+        // Determine how the fields of a & b extend
+        const a_has_uniq_fields = partitioned.only_in_a.len() > 0;
+        const b_has_uniq_fields = partitioned.only_in_b.len() > 0;
+
+        var fields_ext: FieldsExtension = .exactly_the_same;
+        if (a_has_uniq_fields and b_has_uniq_fields) {
+            fields_ext = .both_extend;
+        } else if (a_has_uniq_fields) {
+            fields_ext = .a_extends_b;
+        } else if (b_has_uniq_fields) {
+            fields_ext = .b_extends_a;
+        }
+
+        // Unify fields
+        switch (fields_ext) {
+            .exactly_the_same => {
+                // Unify exts (these will both be the empty record)
+                try self.unifyGuarded(a_gathered_fields.ext, b_gathered_fields.ext);
```

Comment:
wondering if the comment is not entirely correct - if we unify `{a: Int}b ~ {a: Int}c`, the ext will be a flex var right?